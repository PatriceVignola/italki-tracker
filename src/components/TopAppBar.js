/**
 * @prettier
 * @flow
 */

import React from 'react';
import {AppBar, Toolbar, IconButton, Typography} from '@material-ui/core';

import MenuIcon from '@material-ui/icons/Menu';

type Props = {
  onHamburgerClick: () => void,
};

function TopAppBar(props: Props) {
  return (
    <AppBar position="static">
      <Toolbar>
        <IconButton
          style={styles.menuButton}
          onClick={props.onHamburgerClick}
          color="inherit"
          aria-label="Menu"
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="title" color="inherit" style={styles.title}>
          Italki Tracker
        </Typography>
      </Toolbar>
    </AppBar>
  );
}

const styles = {
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    flex: 1,
  },
};

export default TopAppBar;
