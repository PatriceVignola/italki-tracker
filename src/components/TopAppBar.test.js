/**
 * @prettier
 * @flow
 */

import React from 'react';
import {shallow} from 'enzyme';

import TopAppBar from './TopAppBar';

describe('TopAppBar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TopAppBar onHamburgerClick={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
