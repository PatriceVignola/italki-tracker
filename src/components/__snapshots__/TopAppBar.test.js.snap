// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAppBar renders correctly 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Menu"
      color="inherit"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": -12,
          "marginRight": 20,
        }
      }
    >
      <pure(Menu) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      color="inherit"
      style={
        Object {
          "flex": 1,
        }
      }
      variant="title"
    >
      Italki Tracker
    </WithStyles(Typography)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
